pkg/alertmanager/statefulset.go

```diff
@@ -306,6 +306,13 @@ func makeStatefulSetSpec(a *monitoringv1.Alertmanager, config Config, tlsAssetSe
 		amArgs = append(amArgs, fmt.Sprintf("--cluster.peer-timeout=%s", a.Spec.ClusterPeerTimeout))
 	}
 
+	// --cluster.label flag was introduced in alertmanager v0.26, this helps to block any traffic that
```

Comment:
(nit) let's add more context about the issue being fixed.
```suggestion
	// If multiple Alertmanager clusters are deployed on the same cluster, it can happen that because pod IP addresses are recycled, an Alertmanager instance from cluster B connects with cluster A.
	// --cluster.label flag was introduced in alertmanager v0.26 and prevents an instance to connect to other cluster's member unless they share the same label.
```